File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
from sys import stdin as s

N = int(s.readline())
arr =[]
for i in range(N):
    A,B = map(int,s.readline().split())
    arr.append((A,B))

arr.sort(key=lambda x:(x[1],x[0]))
cnt =1 
ans = arr[0][1]
for i in range(1,N):    
    if ans<=arr[i][0]:
        cnt+=1
        ans=arr[i][1]
        
    
print(cnt)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.StringTokenizer;

public class Main {

	public static void main(String[] args) throws IOException {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));


		int N = Integer.parseInt(br.readLine());
		int[][] arr = new int[N][2];

		for (int i = 0; i < N; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			arr[i][0] = Integer.parseInt(st.nextToken());
			arr[i][1] = Integer.parseInt(st.nextToken());
		}

		Arrays.sort(arr, ((o1, o2) -> {
			if (o1[1] == o2[1]) {
				return o1[0] - o2[0];
			}
			return o1[1] - o2[1];
		}));

		int answer = 1;
		int end = arr[0][1];
		for (int j = 1; j < N; j++) {
			if (arr[j][0] >= end) {
				end = arr[j][1];
				answer++;
			}
		}

		System.out.println(answer);

	}
}
